Feat(CORE-941): Add customizing modules section (#845)
gruntwork-io/docs: _docs-sources/iac/usage/customizing-modules.md

# Customizing Modules

> We strive to keep the Infrastructure as Code Library flexible and unopinionated, however there may be instances where you need to customize a module to your specific use case or a company policy that does not allow you to pull code from an external source.

Reviewer: ```suggestion
We strive to keep the Infrastructure as Code Library flexible and generic enough to serve common use cases, however there may be instances where you need to customize a module to your specific use case or a company policy that does not allow you to pull code from an external source.
```

Reviewer: NIT: I think `unopinionated` is particularly problematic given that we advertise being opinionated as a key feature throughout the rest of our marketing materials. This could change rapidly, but feels like it would be jarring to folks paying attention.

Author: Good call! I lifted this from some existing docs but felt not-so-great about the usage of `unopinionated`.

> We strive to keep the Infrastructure as Code Library flexible and unopinionated, however there may be instances where you need to customize a module to your specific use case or a company policy that does not allow you to pull code from an external source.

Reviewer: ```suggestion
We strive to keep the Infrastructure as Code Library flexible and unopinionated, however there may be instances where you need to customize a module to your specific use case, or ensure compliance with a company policy that does not allow you to pull code from an external source.
```

Author: Added in [1bb5e8c](https://github.com/gruntwork-io/docs/pull/845/commits/1bb5e8c1cde7347864f815c74d604697d25dff36)

> We strive to keep the Infrastructure as Code Library flexible and generic enough to serve common use cases, however there may be instances where you need to customize a module to your specific use case, or to ensure compliance with a company policy that does not allow you to pull code from an external source.
> 
> In these cases, you may either fork the code to your own organization or create a pull request that implements the desired functionality.

Reviewer 2: ```suggestion
We strive to keep the Infrastructure as Code Library flexible and generic enough to serve common use cases, however there may be instances where you need to customize a module to your specific use case, or to ensure compliance with a company policy that does not allow you to pull code from an external source. In these cases, you may either fork the code to your own organization or create a pull request that implements the desired functionality.
```

## Forking

:::caution

> Whenever possible, we strongly recommend using the code directly from the gruntwork-io GitHub org and avoid forking. If your company completely bans all outside sources, then follow the instructions below.

Author: Added in [1bb5e8c](https://github.com/gruntwork-io/docs/pull/845/commits/1bb5e8c1cde7347864f815c74d604697d25dff36)

> Whenever possible, we strongly recommend using the code directly from the gruntwork-io GitHub org and avoid forking. If your company completely bans all outside sources, then follow the instructions below.

Reviewer 3: ```suggestion
Whenever possible, we strongly recommend using the code directly from the `gruntwork-io` GitHub org and avoid forking. If your company completely bans all outside sources, then follow the instructions below.
```

> Whenever possible, we strongly recommend using the code directly from the `gruntwork-io` GitHub org and avoid forking due to the [increased overhead of managing the fork](#drawbacks-to-forking). If your company completely bans all outside sources, then follow the instructions below.

Reviewer 2: ```suggestion
Whenever possible, we strongly recommend that you use code directly from the `gruntwork-io` GitHub org and avoid forking due to the [increased overhead of managing the fork](#drawbacks-to-forking). If your company completely bans all outside sources, then follow the instructions below.
```

:::

> The [Gruntwork Terms of Service](https://gruntwork.io/terms/) give you permissions to fork the code from the Gruntwork Infrastructure as Code Library into your own repos. If your company does not allow external dependencies (e.g., you have a company policy that requires all source code to be pulled from an internal GitHub Enterprise or BitBucket server) or if you need to make modifications to the Infrastructure as Code Library that you do not wish to contribute back to Gruntwork. This section will walk you through what you need to do to fork the code.

Reviewer: NIT: second to last sentence is a fragment if we don't combine these two.

> The [Gruntwork Terms of Service](https://gruntwork.io/terms/) give you permissions to fork the code from the Gruntwork Infrastructure as Code Library into your own repos. If your company does not allow external dependencies (e.g., you have a company policy that requires all source code to be pulled from an internal GitHub Enterprise or BitBucket server) or if you need to make modifications to the Infrastructure as Code Library that you do not wish to contribute back to Gruntwork. This section will walk you through what you need to do to fork the code.

Author: I'm going to flip it and reverse it to see if that reads nicer

> The [Gruntwork Terms of Service](https://gruntwork.io/terms/) give you permissions to fork the code from the Gruntwork Infrastructure as Code Library into your own repos. If your company does not allow external dependencies (e.g., you have a company policy that requires all source code to be pulled from an internal GitHub Enterprise or BitBucket server) or if you need to make modifications to the Infrastructure as Code Library that you do not wish to contribute back to Gruntwork. This section will walk you through what you need to do to fork the code.

Author: See [1bb5e8c](https://github.com/gruntwork-io/docs/pull/845/commits/1bb5e8c1cde7347864f815c74d604697d25dff36) for change
